File: scraper_service/sources/base_source.py
from abc import ABC, abstractmethod


class BaseSource(ABC):
    @abstractmethod
    def make_query_url(self, keyword, page):
        pass

    @abstractmethod
    def parse_job_list_page(self, keyword, soup):
        pass

    @abstractmethod
    def parse_job_detail(self, keyword, job):
        pass

File: scraper_service_js/sources/base_source.js
class BaseSource {
  // Define the abstract methods
  sourceUrl(keyword, page) {
    throw new Error('Method "sourceUrl" must be implemented.');
  }

  parseSourceJob(baseUrl, html = null, job = null) {
    throw new Error('Method "parseSourceJob" must be implemented.');
  }
}

module.exports = BaseSource;
